lib/src/json_schema/json_schema.dart

```diff
@@ -668,6 +668,8 @@ class JsonSchema {
   /// The parent [JsonSchema] for this [JsonSchema].
   JsonSchema _parent;
 
+  List<JsonSchema> _dynamicParents = [];
```

Comment:
> Lexical and dynamic scopes align until a reference keyword is encountered. While following the reference keyword moves processing from one lexical scope into a different one, from the perspective of dynamic scope, following reference is no different from descending into a subschema present as a value. A keyword on the far side of that reference that resolves information through the dynamic scope will consider the originating side of the reference to be their dynamic parent, rather than examining the local lexically enclosing parent.

Might be nice to reference https://json-schema.org/draft/2019-09/json-schema-core.html#rfc.section.7.1.

Reply:
And the fact we're going to use this as a stack.